editor/import/resource_importer_imagefont.cpp

```diff
@@ -253,7 +251,12 @@ Error ResourceImporterImageFont::import(const String &p_source_file, const Strin
 				}
 			}
 			if (end == -1) {
-				end = start;
+				if (step == STEP_END_READ_HEX) {
+					end = token.hex_to_int();
+				}
+				if (step == STEP_END_READ_DEC) {
+					end = token.to_int();
+				}
 			}
```

Comment:
This should check for all decode condition (both start and end), since it not always a range:
```c++
if (step == STEP_START_READ_HEX) {
	start = token.hex_to_int();
} else if (step == STEP_START_READ_DEC) {
	start = token.to_int();
} else if (step == STEP_END_READ_HEX) {
	end = token.hex_to_int();
} else if (step == STEP_END_READ_DEC) {
	end = token.to_int();
}

if (end == -1) {
	end = start;
}
```